### DISCUSSION TITLE
Making fft work with magick windows build, no recompile

### DISCUSSION THREAD

**Discussion by andrixnet** [ImageMagick/ImageMagick#6776] [Help] [answered]
### ImageMagick version

7.1.1-20

### Operating system, version and so on

Win7

### Description

I'm trying to run some FFT processing commands with ImageMagick, starting with the basic tests like `convert lena.png -fft -ift lena_mp_roundtrip.png`
First I got the error of missing FFTW library.
I head over to http://www.fftw.org/install/windows.html and get the DLLs, which I copy  to c:\windows\system32, but are not recognized. I move them to the imagemagick directory, still they are not recognized. 
The `bench` utility from FFTW works. 

This is the error I get: 
```
d:\work>magick lena.png -fft -ift lena_mp_roundtrip.png
magick: delegate library support not built-in `lena.png' (FFTW) @ warning/fourier.c/ForwardFourierTransformImage/922.
magick: image sequence is required `-ift' @ error/operation.c/CLIListOperatorImages/4340.
```

This is the version information:
```
d:\work>magick -version
Version: ImageMagick 7.1.1-20 Q16-HDRI x64 98bb1d4:20231008 https://imagemagick.
org
Copyright: (C) 1999 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Channel-masks(64-bit) Cipher DPC HDRI Modules OpenCL OpenMP(2.0)
Delegates (built-in): bzlib cairo flif freetype gslib heic jng jp2 jpeg jxl lcms lqr lzma openexr pangocairo png ps raqm raw rsvg tiff webp xml zip zlib
Compiler: Visual Studio 2022 (193532217)
```

I downloaded the current windows binary release (static) https://imagemagick.org/script/download.php#windows

Unfortunately building from source on Windows is way beyond my skills. :-(

**Comment by snibgo** [comment]
When `magick -version` shows that fftw isn't a built-in delegate, the only way to get it is by rebuilding IM. Merely copying a fftw library from somewhere won't work, because IM contains code that is compiled only when the fftw delegate is enabled. The IM source for Windows includes fftw and other common libraries, so the process is quite simple.

**Comment by dlemstra** [comment]
The fftw library is not included because of the license that it uses. They use the GPL license and that is incompatible with out Apache 2.0 license. You will need to build from source if you would want to include this.

**Comment by andrixnet** [comment]
I've seen building instructions using Visual Studio (which I don't have access to). I've also read fragments about compilation with alternate compiler suites such as mingw or cygwin, but few details and posts from many years ago. 
Can you please point me to some more recent and more detaild docs to try with these?
Thanks.

**Comment by snibgo** [comment, marked as answer]
I build IM on Windows with the Gnu toolchain from Cygwin.

For *using* IM with Cygwin, see my page [IM with Cygwin](http://im.snibgo.com/cygwin.htm).

For *building* IM with Cygwin bash, see [Compiling IM with Cygwin: Building from Github](http://im.snibgo.com/compim.htm#github).
